hosted/eks/helper/helper_cluster.go

```diff
@@ -3,83 +3,204 @@ package helper
+		Eventually(func() string {
+			ginkgo.GinkgoLogr.Info("Waiting for k8s upgrade to appear in EKSStatus.UpstreamSpec ...")
+			cluster, err = client.Management.Cluster.ByID(cluster.ID)
+			Expect(err).To(BeNil())
+			return *cluster.EKSStatus.UpstreamSpec.KubernetesVersion
+		}, tools.SetTimeout(15*time.Minute), 10*time.Second).Should(Equal(upgradeToVersion))
+		// ensure nodegroup version is same in Rancher
+		for _, ng := range cluster.EKSStatus.UpstreamSpec.NodeGroups {
+			Expect(*ng.Version).To(Equal(currentVersion))
+		}
+	}
+	return cluster, nil
+}
+
+// UpgradeNodeKubernetesVersion upgrades the k8s version of nodegroup to the value defined by upgradeToVersion.
+// if wait is set to true, it will wait until the cluster finishes upgrading;
+// if checkClusterConfig is set to true, it will validate that nodegroup has been upgraded successfully
+func UpgradeNodeKubernetesVersion(cluster *management.Cluster, upgradeToVersion string, client *rancher.Client, wait, checkClusterConfig bool) (*management.Cluster, error) {
+	upgradedCluster := cluster
+	for i := range upgradedCluster.EKSConfig.NodeGroups {
+		upgradedCluster.EKSConfig.NodeGroups[i].Version = &upgradeToVersion
+	}
+
+	var err error
+	cluster, err = client.Management.Cluster.Update(cluster, &upgradedCluster)
+	Expect(err).To(BeNil())
+
+	// Check if the desired config is set correctly
+	for _, ng := range cluster.EKSConfig.NodeGroups {
+		Expect(*ng.Version).To(Equal(upgradeToVersion))
+	}
+
+	if wait {
+		err = clusters.WaitClusterToBeUpgraded(client, cluster.ID)
+		Expect(err).To(BeNil())
+	}
+
+	// TODO: Fix flaky check
+	if checkClusterConfig {
+		Eventually(func() bool {
+			ginkgo.GinkgoLogr.Info("waiting for the nodegroup upgrade to appear in EKSStatus.UpstreamSpec ...")
+			// Check if the desired config has been applied in Rancher
+			for _, ng := range cluster.EKSStatus.UpstreamSpec.NodeGroups {
+				if *ng.Version != upgradeToVersion {
+					return false
+				}
+			}
+			return true
+		}, tools.SetTimeout(15*time.Minute), 10*time.Second).Should(BeTrue())
+	}
+	return cluster, nil
+}
+
 // AddNodeGroup adds a nodegroup to the list
-func AddNodeGroup(cluster *management.Cluster, increaseBy int, client *rancher.Client) (*management.Cluster, error) {
-	upgradedCluster := new(management.Cluster)
-	upgradedCluster.Name = cluster.Name
-	upgradedCluster.EKSConfig = cluster.EKSConfig
-	nodeConfig := EksHostNodeConfig()
+// if checkClusterConfig is set to true, it will validate that nodegroup has been added successfully
+func AddNodeGroup(cluster *management.Cluster, increaseBy int, client *rancher.Client, wait, checkClusterConfig bool) (*management.Cluster, error) {
+	upgradedCluster := cluster
+	currentNodeGroupNumber := len(cluster.EKSConfig.NodeGroups)
+
+	// Workaround for eks-operator/issues/406
+	var eksClusterConfig management.EKSClusterConfigSpec
+	config.LoadConfig(eks.EKSClusterConfigConfigurationFileKey, &eksClusterConfig)
```

Comment:
This might not be needed since we assign the following workaround after importing a cluster:
```go
// Workaround to add new Nodegroup till https://github.com/rancher/aks-operator/issues/251 is fixed
cluster.EKSConfig = cluster.EKSStatus.UpstreamSpec
```

Reply:
Tested, and seems required as of now